src/components/SidebarLayout/WalletInfo/index.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react'
+import styled from 'styled-components'
+import { getNetwork } from 'src/config'
+
+import {
+  Icon,
+  FixedIcon,
+  EthHashInfo,
+  Text,
+  Identicon,
+  Button,
+  CopyToClipboardBtn,
+  EtherscanButton,
+} from '@gnosis.pm/safe-react-components'
+
+export const TOGGLE_SIDEBAR_BTN_TESTID = 'TOGGLE_SIDEBAR_BTN'
+
+const Container = styled.div`
+  max-width: 200px;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+`
+
+const IdenticonContainer = styled.div`
+  width: 100%;
+  margin: 8px;
+  display: flex;
+  justify-content: space-between;
+
+  div:first-of-type {
+    width: 32px;
+  }
+`
+
+const IconContainer = styled.div`
+  width: 100px;
+  display: flex;
+  padding: 8px 0;
+  justify-content: space-evenly;
+`
+const StyledButton = styled(Button)`
+  *:first-child {
+    margin: 0 4px 0 0;
+  }
+`
+const StyledLabel = styled.div`
+  background-color: ${({ theme }) => theme.colors.icon};
+  margin: 8px 0 0 0 !important;
+  padding: 4px 8px;
+  border-radius: 4px;
+  letter-spacing: 1px;
+`
+const StyledText = styled(Text)`
+  margin: 8px 0 16px 0;
+`
+const UnStyledButton = styled.button`
+  background: none;
+  color: inherit;
+  border: none;
+  padding: 0;
+  font: inherit;
+  cursor: pointer;
+  outline-color: ${({ theme }) => theme.colors.separator};
+  display: flex;
+  align-items: center;
+`
+
+type Props = {
+  address: string | null
+  safeName: string
+  granted: boolean
+  balance: string | null
+  onToggleSafeList: () => void
+  onReceiveClick: () => void
+  onNewTransactionClick: () => void
+}
+
+const WalletInfo = ({
+  address,
+  safeName,
+  balance,
+  granted,
+  onToggleSafeList,
+  onReceiveClick,
+  onNewTransactionClick,
+}: Props): React.ReactElement => {
+  if (!address) {
+    return (
+      <Container>
+        <IdenticonContainer>
+          <div></div>
+          <div>
+            <FixedIcon type="notConnected" />
+          </div>
+          <UnStyledButton onClick={onToggleSafeList} data-testid={TOGGLE_SIDEBAR_BTN_TESTID}>
+            <Icon size="md" type="circleDropdown" />
+          </UnStyledButton>
+        </IdenticonContainer>
+        NO SAFE LOADED
+      </Container>
+    )
+  }
+
+  return (
+    <Container>
+      <IdenticonContainer>
+        <div></div>
```

Comment:
What is this empty `div` used for? Same as before

Reply:
```
  display: flex;
  justify-content: space-between;
``` 
that's to make `justify-content` work.

Reply:
replaced by `<FlexSpacer />`